crates/nu-command/src/filters/drop/nth.rs

```diff
@@ -102,108 +99,137 @@ impl Command for DropNth {
+                                    return Err(ShellError::UnsupportedInput {
+                                        msg: "The upper bound must be greater than the lower bound"
+                                            .into(),
+                                        input: "value originates from here".into(),
+                                        msg_span: head,
+                                        input_span: value.span(),
+                                    });
+                                }
 
-                (start..=end).collect()
+                                let end = if let PipelineData::Value(Value::List { vals, .. }, _) =
+                                    &input
+                                {
+                                    (end - 1).min((vals.len() as i64) - 1)
+                                } else {
+                                    end - 1
+                                };
+
+                                rows_to_drop.extend((start as usize)..=(end as usize));
+                            }
+                            Bound::Unbounded => {
+                                let start_usize = start as usize;
+                                min_unbounded_start = Some(
+                                    min_unbounded_start.map_or(start_usize, |s| s.min(start_usize)),
+                                );
+                            }
+                        }
+                    }
+                    Range::FloatRange(_) => {
+                        return Err(ShellError::UnsupportedInput {
+                            msg: "float range not supported".into(),
+                            input: "value originates from here".into(),
+                            msg_span: head,
+                            input_span: value.span(),
+                        });
+                    }
+                }
+            } else {
+                return Err(ShellError::TypeMismatch {
+                    err_message: "Expected int or range".into(),
+                    span: value.span(),
+                });
             }
+        }
+
+        rows_to_drop.sort_unstable();
+        rows_to_drop.dedup();
+
+        let input = if let Some(cutoff) = min_unbounded_start {
+            input
+                .into_iter()
+                .take(cutoff)
+                .into_pipeline_data_with_metadata(
+                    head,
+                    engine_state.signals().clone(),
+                    metadata.clone(),
+                )
+        } else {
+            input
         };
 
         Ok(DropNthIterator {
             input: input.into_iter(),
-            rows,
+            rows: rows_to_drop,
             current: 0,
         }
-        .into_pipeline_data_with_metadata(head, engine_state.signals().clone(), metadata))
+        .into_pipeline_data_with_metadata(
+            head,
+            engine_state.signals().clone(),
+            metadata.clone(),
```

Comment:
this clone isn't necessary
```suggestion
            metadata,
```